File: Data Structures & Algorithms/binary-search/submission-8.cs
public class Solution {
    public int Search(int[] nums, int target)
    {
        var r = 0;
        var l = nums.Count() - 1;
        var m = 0;

        while (r <= l)
        {
            m = (r + l) / 2;
            if (target == nums[m])
                break;
            else if (target > nums[m])
                r = m + 1;
            else
                l = m - 1;
        }
        if (nums[m] != target) return -1;
        
        return m;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-8.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1

        while(l < r):
            m = l + ((r-l)//2)

            if nums[m] > nums[r]:
                l = m +1
            else:
                r = m

        pivot = l

        if pivot == 0:
            l, r = 0, len(nums) - 1
        elif nums[pivot] <= target <= nums[-1]:
            l, r = pivot, len(nums) - 1
        else:
            l, r = 0, pivot - 1

        while(l <= r):
            m = l + ((r-l)//2)

            if m == len(nums):
                break 

            if target == nums[m]:
                return m
            elif target > nums[m]:
                l = m+1
            else:
                r = m-1
        
        return -1